frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -59,44 +69,176 @@ export async function action({ context, request }: Route.ActionArgs) {
+                v.integer(t('protected:primary-identity-document.citizenship-date.invalid-month')),
+                v.minValue(1, t('protected:primary-identity-document.citizenship-date.invalid-month')),
+                v.maxValue(12, t('protected:primary-identity-document.citizenship-date.invalid-month')),
+              ),
+              citizenshipDateDay: v.pipe(
+                v.number(t('protected:primary-identity-document.citizenship-date.required-day')),
+                v.integer(t('protected:primary-identity-document.citizenship-date.invalid-day')),
+                v.minValue(1, t('protected:primary-identity-document.citizenship-date.invalid-day')),
+                v.maxValue(31, t('protected:primary-identity-document.citizenship-date.invalid-day')),
+              ),
+              citizenshipDate: v.date(),
+              document: v.pipe(
+                v.file(t('protected:primary-identity-document.upload-document.required')),
+                v.mimeType(
+                  ['image/jpeg', 'image/png', 'image/heic'],
+                  t('protected:primary-identity-document.upload-document.invalid'),
+                ),
+                v.maxSize(maxImageSizeBits),
+              ),
+            }),
+          ],
+          t('protected:primary-identity-document.document-type.required'),
         ),
-      ) satisfies v.GenericSchema<PrimaryDocumentsSessionData>;
+      ]) satisfies v.GenericSchema<PrimaryDocumentsSessionData>;
+
+      const dateOfBirthYear = Number(formData.get('dateOfBirthYear'));
+      const dateOfBirthMonth = Number(formData.get('dateOfBirthMonth'));
+      const dateOfBirthDay = Number(formData.get('dateOfBirthDay'));
+
+      const citizenshipDateYear = Number(formData.get('citizenshipDateYear'));
+      const citizenshipDateMonth = Number(formData.get('citizenshipDateMonth'));
+      const citizenshipDateDay = Number(formData.get('citizenshipDateDay'));
 
       const input = {
         currentStatusInCanada: String(formData.get('currentStatusInCanada')),
         documentType: String(formData.get('documentType')),
+        registrationNumber: String(formData.get('registrationNumber')),
+        clientNumber: String(formData.get('clientNumber')),
+        givenName: String(formData.get('givenName')),
+        lastName: String(formData.get('lastName')),
+        dateOfBirthYear: dateOfBirthYear,
+        dateOfBirthMonth: dateOfBirthMonth,
+        dateOfBirthDay: dateOfBirthDay,
+        dateOfBirth: new Date(dateOfBirthYear, dateOfBirthMonth - 1, dateOfBirthDay), // The month as a number between 0 and 11 (January to December)
+        gender: String(formData.get('gender')),
+        citizenshipDateYear: citizenshipDateYear,
+        citizenshipDateMonth: citizenshipDateMonth,
+        citizenshipDateDay: citizenshipDateDay,
+        citizenshipDate: new Date(citizenshipDateYear, citizenshipDateMonth - 1, citizenshipDateDay), // The month as a number between 0 and 11 (January to December)
```

Comment:
This should be represented as a string, in my opinion.. Or, if you really must store it as a date, then it should be normalized to eastern time.. otherwise it will be the birthdate at `T00:00:00Z` which isn't correct.